### PR TITLE
doc: releases: Build system: Introduction of LTO

### PR DESCRIPTION
**Pull request by rakons**
This commit adds documentation about added functionality for now interrupt tables parser and Link Time Optimisation.

### CODE REVIEW

**Comment by microbuilder** [comment]
Related PR for context: https://github.com/zephyrproject-rtos/zephyr/pull/66392

File: doc/releases/release-notes-3.6.rst [hunk 1]
```diff
@@ -149,6 +149,16 @@ Boards & SoC Support
...
+* Added functionality for Link Time Optimisation.
+  This change includes interrupt script generator rebuilding and adding following options:
+
+  - :kconfig:option:`CONFIG_ISR_TABLES_LOCAL_DECLARATION` Kconfig option:
+  LTO compatible interrupt tables parser,
+  - :kconfig:option:`CONFIG_LTO` Kconfig option: Enable Link Time Optimisation.
+
+  Currently the LTO compatible interrupt tables parser is only supported by ARM architectures and
+  GCC compiler/linker.
+
```
Comment (microbuilder) [inline doc/releases/release-notes-3.6.rst:161]: It's probably worth adding a mention of the PR number here for anyone interested in the details (https://github.com/zephyrproject-rtos/zephyr/pull/66392)

Comment (rakons) [reply doc/releases/release-notes-3.6.rst]: Done.

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by microbuilder — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]

File: doc/releases/release-notes-3.6.rst [hunk 2]
```diff
@@ -149,6 +149,17 @@ Boards & SoC Support
...
+* Added functionality for Link Time Optimisation.
+  This change includes interrupt script generator rebuilding and adding following options:
+
+  - :kconfig:option:`CONFIG_ISR_TABLES_LOCAL_DECLARATION` Kconfig option:
+    LTO compatible interrupt tables parser,
+  - :kconfig:option:`CONFIG_LTO` Kconfig option: Enable Link Time Optimisation.
+
+  Currently the LTO compatible interrupt tables parser is only supported by ARM architectures and
+  GCC compiler/linker.
+  See `pull request 66392 <https://github.com/zephyrproject-rtos/zephyr/pull/66392>`_ for details.
```
Comment (kartben) [inline doc/releases/release-notes-3.6.rst:161]: hardcoding links in the documentation should be avoided. 

```suggestion
  See pull request :github:`66392` for details.
```

File: doc/releases/release-notes-3.6.rst [hunk 3]
```diff
@@ -149,6 +149,17 @@ Boards & SoC Support
 Build system and infrastructure
 *******************************
 
+* Added functionality for Link Time Optimisation.
+  This change includes interrupt script generator rebuilding and adding following options:
+
+  - :kconfig:option:`CONFIG_ISR_TABLES_LOCAL_DECLARATION` Kconfig option:
+    LTO compatible interrupt tables parser,
+  - :kconfig:option:`CONFIG_LTO` Kconfig option: Enable Link Time Optimisation.
```
Comment (kartben) [inline doc/releases/release-notes-3.6.rst:157]: nit: American English form (but we don't have a strict rule here --maybe we should-- so feel free to ignore :))

```suggestion
* Added functionality for Link Time Optimization.
  This change includes interrupt script generator rebuilding and adding following options:

  - :kconfig:option:`CONFIG_ISR_TABLES_LOCAL_DECLARATION` Kconfig option:
    LTO compatible interrupt tables parser,
  - :kconfig:option:`CONFIG_LTO` Kconfig option: Enable Link Time Optimization.
```

**Review by kartben — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-3.6.rst (hunk 2 above)
Comment (rakons) [reply doc/releases/release-notes-3.6.rst]: Done

File: doc/releases/release-notes-3.6.rst (hunk 3 above)
Comment (rakons) [reply doc/releases/release-notes-3.6.rst]: Done

**Review by henrikbrixandersen — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by microbuilder — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**